### PR TITLE
Remove unnecessary features and add FEATURE_TYPE_INFO

### PR DESCRIPTION
**Pull request by gpetrou**

### CODE REVIEW

File: Src/IronPython.Modules/time.cs [hunk 1]
```diff
@@ -71,7 +71,6 @@ public static class PythonTime {
 
             // altzone, timezone are offsets from UTC in seconds, so they always fit in the
             // -13*3600 to 13*3600 range and are safe to cast to ints
-#if FEATURE_TIMEZONE
```
Comment (slozier) [inline Src/IronPython.Modules/time.cs]: You want to get rid of the TimeZone stuff and keep the TimeZoneInfo stuff. See IronLanguages/ironpython2#203.

Comment (gpetrou) [inline Src/IronPython.Modules/time.cs]: Oh, so we actually want to remove the code that is currently used?

Comment (slozier) [inline Src/IronPython.Modules/time.cs]: In this case yes.

Comment (gpetrou) [inline Src/IronPython.Modules/time.cs]: OK

**Comment by slide** [comment]
@slozier You good with this one now?

**Pull request merged**